### PR TITLE
feat: add logging aspect jdk 11

### PR DESCRIPTION
**Pull request by caiopavanelli**
# Pull Request Merge Checklist
           
Task backlog identifier related to this PR: ---

---

## As a DEVELOPER, I confirm that this PR **doesn't contain**:
- [x] Code lines with syntax errors
- [x] Methods and variables names that aren't clear or doesn't mean anything
- [x] Code that is out of the standard (Example: **PEP-8** for Python code)
- [x] Files with **imports** errors or without any utility
- [x] **TODOs** left behind
- [x] Tests that are being skipped
- [x] Methods placed on incorrect files or folders (Example: methods that are local, but should be placed on /utils)
- [x] Functions, methods or API responses that are publicly returning sensitive information

## As a DEVELOPER, I confirm that I:
- [x] Followed the Simbiose's standard for branch and commit naming
- [x] Concluded the task and achieved the goals according to the task planning I defined before starting to code
- [x] Added all necessary database changes to the [SQL Migrations](https://github.com/s1mbi0se/sql-migrations)
- [x] Created tests for every new flow and method on this Pull Request
- [x] Added the necessary DEBUG logs
- [x] Checked my code using the Simbiose's Corner Cases Guide
- [x] Merged the **develop** branch to this PR/branch before sending the PR for review
- [x] Solved all possible merge conflicts before sending the PR for review
- [x] Corrected all the problems and code smells identified by the quality tools
- [x] Manually executed the necessary tests locally to make sure my changes didn't break anything
- [x] All unitary and integration tests ran by the CI system are passing
- [x] Check the performance of my changes or additions and they didn't make the system slower (I profiled the code)
- [x] Updated the necessary documentation (Example: OpenAPI, Docstrings, Metadata Database Modeling)
- [x] Made the self-review of the code before sending the PR for review
- [x] Sent this PR for a peer-review
- [x] Made all the corrections appointed during the peer-review
- [x] Confirmed that this PR merge will not cause other systems or dependencies to stop working (Example: ShannonDB Python Client and ShannonDB)
- [x] Checked all GitHub Actions related to this PR that can block my merge

---  

## As the TEAM MANAGER, I confirm that I:
- [x] Checked that the code was developed following the best practices for the language, with the best possible readability and using the less possible quantity of code lines and complexity. 
- [x] Checked if the Pull Request and commit messages follow the Simbiose’s defined standard 
- [x] Checked if there is no possible merge conflicts before reviewing the PR
- [x] Checked if the purpose of the task was reached with this PR
- [x] Checked that the code was developed following the best practices for the language, with the best possible readability and using the less possible quantity of code lines and complexity. 
- [x] Checked if all related PRs were created when other components were affected by this task (e.g. database changes on [SQL Migrations](https://github.com/s1mbi0se/sql-migrations))
- [x] Checked that were created unit, integration and chaos tests for every new flow and method on this Pull Request
- [x] Checked that no sensible information (e.g. users security data, internal security data, etc) was exposed by this PR
- [x] Checked that code coverage for this PR is higher than 80%.
- [x] Checked that there are no problems and code smells identified by the quality and security tools
- [x] Checked that all files were created on the proper folders according to the Simbiose’s defined standard
- [x] Checked that all classes, methods and variables on this PR follow the Simbiose’s pattern for this programming language
- [x] Checked that all unitary and integration tests ran by the CI system are passing
- [x] Checked that there are no performance issues created by these changes or additions
- [x] Checked that all documentation is up to date according to this change (Example: OpenAPI, Docstrings, Metadata Database Modeling)
- [x] Checked that this PR merge will not cause other systems or dependencies to stop working (Example: ShannonDB Python Client and ShannonDB)
- [x] Checked all GitHub Actions related to this PR that can block the merge

---  

## As the QUALITY RESPONSIBLE, I confirm that I:
- [ ] Checked if the Pull Request and commit messages follow the Simbiose’s defined standard 
- [ ] Checked if there is no possible merge conflicts before reviewing the PR
- [ ] Checked that were created unit, integration and chaos tests for every new flow and method on this Pull Request
- [ ] Checked that no sensible information (e.g. users security data, internal security data, etc) was exposed by this PR
- [ ] Checked that code coverage for this PR is higher than 80%.
- [ ] Checked that there are no problems and code smells identified by the quality and security tools
- [ ] Checked that all files were created on the proper folders according to the Simbiose’s defined standard
- [ ] Checked that all classes, methods and variables on this PR follow the Simbiose’s code standard for this programming language
- [ ] Checked that all unitary and integration tests ran by the CI system are passing
- [ ] Checked that there are no performance issues created by these changes or additions
- [ ] Checked that all documentation is up to date according to this change (Example: OpenAPI, Docstrings, Metadata Database Modeling)
- [ ] Checked that this PR merge will not cause other systems or dependencies to stop working (Example: ShannonDB Python Client and ShannonDB)
- [ ] Checked all GitHub Actions related to this PR that can block the merge

## Important links:
1. [Simbiose Quality Guidelines](https://github.com/s1mbi0se/company/wiki/Quality-Topics#table-of-contents)
2. [Metadata Database Modeling](https://docs.google.com/spreadsheets/d/1Y8_k5xV86kTjgfjbi-PGsJcubUKhWlT2x8ecpnMlyR0/edit?usp=sharing)
3. [Simbiose's Corner Cases Guide](https://www.simbioseventures.com/documentos-e-guias/procedimentos-e-checklists/corner-cases-guide)

### CODE REVIEW

File: presto-main/src/main/java/io/prestosql/logging/aspects/SqlCallFlowLoggingAspect.java [hunk 1]
```diff
@@ -0,0 +1,163 @@
...
+                    "!within(io.prestosql.logging..*)";
+
+    private static final String FINISH_METHOD =
+            "execution(* io.prestosql.execution.QueryStateTimer.endQuery())";
+
+    protected final Map<Long, Integer> threadIdToStep = new ConcurrentHashMap<>();
+    protected final Map<Long, Long> threadIdToDebugLogId = new ConcurrentHashMap<>();
+
+    /**
+     * This method is executed when the Thread starts execution of creation schema command inside
```
Comment (joaopedro-simbiose) [inline presto-main/src/main/java/io/prestosql/logging/aspects/SqlCallFlowLoggingAspect.java:46]: I believe all the javadocs are not correct, this description is from ShannonDB. Please, review all this documentation and adapt it please

File: presto-main/src/main/java/io/prestosql/logging/aspects/SqlCallFlowLoggingAspect.java [hunk 2]
```diff
@@ -0,0 +1,163 @@
...
+    public Object around(final ProceedingJoinPoint point)
+            throws Throwable
+    {
+        final long threadId = Thread.currentThread().getId();
+
+        return printDebugLogForMethod(point, threadId);
+    }
+
+    /**
+     * This method is executed when the Thread starts execution when finishes the command of create
```
Comment (joaopedro-simbiose) [inline presto-main/src/main/java/io/prestosql/logging/aspects/SqlCallFlowLoggingAspect.java:95]: I believe all the javadocs are not correct, this description is from ShannonDB. Please, review all this documentation and adapt it please

File: presto-main/src/main/java/io/prestosql/logging/aspects/SqlCallFlowLoggingAspect.java [hunk 3]
```diff
@@ -0,0 +1,163 @@
...
+        final Object resultFromMethod = printDebugLogForMethod(point, threadId);
+
+        threadIdToStep.remove(threadId);
+        threadIdToDebugLogId.remove(threadId);
+
+        return resultFromMethod;
+    }
+
+    /**
+     * Defines a name for a general flux inside ShannonDB
```
Comment (joaopedro-simbiose) [inline presto-main/src/main/java/io/prestosql/logging/aspects/SqlCallFlowLoggingAspect.java:124]: Do not use the word "flux". Always use flow. Please, search for all the use of the word "flux" and change it please

File: presto-main/src/main/java/io/prestosql/logging/aspects/SqlSelectFlowLoggingAspect.java [hunk 4]
```diff
@@ -0,0 +1,163 @@
...
+                    "!within(io.prestosql.logging..*)";
+
+    private static final String FINISH_METHOD =
+            "execution(* io.prestosql.execution.TaskInfo.isNeedsPlan())";
+
+    protected final Map<Long, Integer> threadIdToStep = new ConcurrentHashMap<>();
+    protected final Map<Long, Long> threadIdToDebugLogId = new ConcurrentHashMap<>();
+
+    /**
+     * This method is executed when the Thread starts execution of creation schema command inside
```
Comment (joaopedro-simbiose) [inline presto-main/src/main/java/io/prestosql/logging/aspects/SqlSelectFlowLoggingAspect.java:46]: I believe all the javadocs are not correct, this description is from ShannonDB. Please, review all this documentation and adapt it please

File: presto-main/src/main/java/io/prestosql/logging/aspects/SqlSelectFlowLoggingAspect.java [hunk 5]
```diff
@@ -0,0 +1,163 @@
...
+    public Object around(final ProceedingJoinPoint point)
+            throws Throwable
+    {
+        final long threadId = Thread.currentThread().getId();
+
+        return printDebugLogForMethod(point, threadId);
+    }
+
+    /**
+     * This method is executed when the Thread starts execution when finishes the command of create
```
Comment (joaopedro-simbiose) [inline presto-main/src/main/java/io/prestosql/logging/aspects/SqlSelectFlowLoggingAspect.java:95]: I believe all the javadocs are not correct, this description is from ShannonDB. Please, review all this documentation and adapt it please

File: presto-main/src/main/java/io/prestosql/logging/aspects/SqlSelectFlowLoggingAspect.java [hunk 6]
```diff
@@ -0,0 +1,163 @@
...
+        final Object resultFromMethod = printDebugLogForMethod(point, threadId);
+
+        threadIdToStep.remove(threadId);
+        threadIdToDebugLogId.remove(threadId);
+
+        return resultFromMethod;
+    }
+
+    /**
+     * Defines a name for a general flux inside ShannonDB
```
Comment (joaopedro-simbiose) [inline presto-main/src/main/java/io/prestosql/logging/aspects/SqlSelectFlowLoggingAspect.java:124]: Use flow instead of flux everywhere

File: presto-main/src/main/java/io/prestosql/logging/aspects/UpdateCatalogsFlow.java [hunk 7]
```diff
@@ -0,0 +1,165 @@
...
+     * @return the name of a general flux inside ShannonDB.
+     */
+    @Override
+    protected String getFlowName()
+    {
+        return FLOW_NAME;
+    }
+
+    /**
+     * Defines a map of a thread to a step(or sequence) for a general flux inside ShannonDB
```
Comment (joaopedro-simbiose) [inline presto-main/src/main/java/io/prestosql/logging/aspects/UpdateCatalogsFlow.java:140]: Use flow instead of flux everywhere please

File: presto-main/src/main/java/io/prestosql/logging/aspects/UpdateCatalogsFlow.java [hunk 8]
```diff
@@ -0,0 +1,165 @@
...
+    public Object around(final ProceedingJoinPoint point)
+            throws Throwable
+    {
+        final long threadId = Thread.currentThread().getId();
+
+        return printDebugLogForMethod(point, threadId);
+    }
+
+    /**
+     * This method is executed when the Thread starts execution when finishes the command of create
```
Comment (joaopedro-simbiose) [inline presto-main/src/main/java/io/prestosql/logging/aspects/UpdateCatalogsFlow.java:97]: I believe all the javadocs are not correct, this description is from ShannonDB. Please, review all this documentation and adapt it please

File: presto-main/src/main/java/io/prestosql/logging/aspects/UpdateCatalogsFlow.java [hunk 9]
```diff
@@ -0,0 +1,165 @@
...
+
+    private static final String FINISH_METHOD =
+            "execution(* io.prestosql.metadata.DynamicCatalogStore"
+                    + ".finishUpdateCatalogDelta(..))";
+
+    protected final Map<Long, Integer> threadIdToStep = new ConcurrentHashMap<>();
+    protected final Map<Long, Long> threadIdToDebugLogId = new ConcurrentHashMap<>();
+
+    /**
+     * This method is executed when the Thread starts execution of creation schema command inside
```
Comment (joaopedro-simbiose) [inline presto-main/src/main/java/io/prestosql/logging/aspects/UpdateCatalogsFlow.java:48]: I believe all the javadocs are not correct, this description is from ShannonDB. Please, review all this documentation and adapt it please

File: presto-main/src/main/java/io/prestosql/logging/aspects/SqlShowFlowLoggingAspect.java [hunk 10]
```diff
@@ -0,0 +1,163 @@
...
+                    "!within(io.prestosql.logging..*)";
+
+    private static final String FINISH_METHOD =
+            "execution(* io.prestosql.execution.TaskInfo.isNeedsPlan())";
+
+    protected final Map<Long, Integer> threadIdToStep = new ConcurrentHashMap<>();
+    protected final Map<Long, Long> threadIdToDebugLogId = new ConcurrentHashMap<>();
+
+    /**
+     * This method is executed when the Thread starts execution of creation schema command inside
```
Comment (joaopedro-simbiose) [inline presto-main/src/main/java/io/prestosql/logging/aspects/SqlShowFlowLoggingAspect.java:46]: I believe all the javadocs are not correct, this description is from ShannonDB. Please, review all this documentation and adapt it please

File: presto-main/src/main/java/io/prestosql/logging/aspects/SqlShowFlowLoggingAspect.java [hunk 11]
```diff
@@ -0,0 +1,163 @@
...
+    public Object around(final ProceedingJoinPoint point)
+            throws Throwable
+    {
+        final long threadId = Thread.currentThread().getId();
+
+        return printDebugLogForMethod(point, threadId);
+    }
+
+    /**
+     * This method is executed when the Thread starts execution when finishes the command of create
```
Comment (joaopedro-simbiose) [inline presto-main/src/main/java/io/prestosql/logging/aspects/SqlShowFlowLoggingAspect.java:95]: I believe all the javadocs are not correct, this description is from ShannonDB. Please, review all this documentation and adapt it please

File: presto-main/src/main/java/io/prestosql/logging/aspects/SqlShowFlowLoggingAspect.java [hunk 12]
```diff
@@ -0,0 +1,163 @@
...
+        final Object resultFromMethod = printDebugLogForMethod(point, threadId);
+
+        threadIdToStep.remove(threadId);
+        threadIdToDebugLogId.remove(threadId);
+
+        return resultFromMethod;
+    }
+
+    /**
+     * Defines a name for a general flux inside ShannonDB
```
Comment (joaopedro-simbiose) [inline presto-main/src/main/java/io/prestosql/logging/aspects/SqlShowFlowLoggingAspect.java:124]: Everywhere, substitute the word flux for flow please

File: presto-main/src/main/java/io/prestosql/logging/aspects/SqlFlowLoggingAspect.java [hunk 13]
```diff
@@ -0,0 +1,164 @@
...
+        final Object resultFromMethod = printDebugLogForMethod(point, threadId);
+
+        threadIdToStep.remove(threadId);
+        threadIdToDebugLogId.remove(threadId);
+
+        return resultFromMethod;
+    }
+
+    /**
+     * Defines a name for a general flux inside ShannonDB
```
Comment (joaopedro-simbiose) [inline presto-main/src/main/java/io/prestosql/logging/aspects/SqlFlowLoggingAspect.java:125]: Not use flux, use flow everywhere please

File: presto-main/src/main/java/io/prestosql/logging/aspects/SqlFlowLoggingAspect.java [hunk 14]
```diff
@@ -0,0 +1,164 @@
...
+    public Object around(final ProceedingJoinPoint point)
+            throws Throwable
+    {
+        final long threadId = Thread.currentThread().getId();
+
+        return printDebugLogForMethod(point, threadId);
+    }
+
+    /**
+     * This method is executed when the Thread starts execution when finishes the command of create
```
Comment (joaopedro-simbiose) [inline presto-main/src/main/java/io/prestosql/logging/aspects/SqlFlowLoggingAspect.java:96]: I believe all the javadocs are not correct, this description is from ShannonDB. Please, review all this documentation and adapt it please

File: presto-main/src/main/java/io/prestosql/logging/aspects/SqlFlowLoggingAspect.java [hunk 15]
```diff
@@ -0,0 +1,164 @@
...
+                    "!within(io.airlift..*)";
+
+    private static final String FINISH_METHOD =
+            "execution(* io.prestosql.execution.QueryStateMachine.transitionToFinishing())";
+
+    protected final Map<Long, Integer> threadIdToStep = new ConcurrentHashMap<>();
+    protected final Map<Long, Long> threadIdToDebugLogId = new ConcurrentHashMap<>();
+
+    /**
+     * This method is executed when the Thread starts execution of creation schema command inside
```
Comment (joaopedro-simbiose) [inline presto-main/src/main/java/io/prestosql/logging/aspects/SqlFlowLoggingAspect.java:47]: I believe all the javadocs are not correct, this description is from ShannonDB. Please, review all this documentation and adapt it please

File: presto-main/src/main/java/io/prestosql/logging/aspects/SqlCallFlowLoggingAspect.java (hunk 1 above)
Comment (caiopavanelli) [reply presto-main/src/main/java/io/prestosql/logging/aspects/SqlCallFlowLoggingAspect.java]: fixed

File: presto-main/src/main/java/io/prestosql/logging/aspects/SqlCallFlowLoggingAspect.java (hunk 2 above)
Comment (caiopavanelli) [reply presto-main/src/main/java/io/prestosql/logging/aspects/SqlCallFlowLoggingAspect.java]: fixed

File: presto-main/src/main/java/io/prestosql/logging/aspects/SqlCallFlowLoggingAspect.java (hunk 3 above)
Comment (caiopavanelli) [reply presto-main/src/main/java/io/prestosql/logging/aspects/SqlCallFlowLoggingAspect.java]: fixed

File: presto-main/src/main/java/io/prestosql/logging/aspects/SqlSelectFlowLoggingAspect.java (hunk 4 above)
Comment (caiopavanelli) [reply presto-main/src/main/java/io/prestosql/logging/aspects/SqlSelectFlowLoggingAspect.java]: fixed

File: presto-main/src/main/java/io/prestosql/logging/aspects/SqlSelectFlowLoggingAspect.java (hunk 5 above)
Comment (caiopavanelli) [reply presto-main/src/main/java/io/prestosql/logging/aspects/SqlSelectFlowLoggingAspect.java]: fixed

File: presto-main/src/main/java/io/prestosql/logging/aspects/SqlSelectFlowLoggingAspect.java (hunk 6 above)
Comment (caiopavanelli) [reply presto-main/src/main/java/io/prestosql/logging/aspects/SqlSelectFlowLoggingAspect.java]: fixed

File: presto-main/src/main/java/io/prestosql/logging/aspects/UpdateCatalogsFlow.java (hunk 7 above)
Comment (caiopavanelli) [reply presto-main/src/main/java/io/prestosql/logging/aspects/UpdateCatalogsFlow.java]: fixed

File: presto-main/src/main/java/io/prestosql/logging/aspects/UpdateCatalogsFlow.java (hunk 8 above)
Comment (caiopavanelli) [reply presto-main/src/main/java/io/prestosql/logging/aspects/UpdateCatalogsFlow.java]: fixed

File: presto-main/src/main/java/io/prestosql/logging/aspects/UpdateCatalogsFlow.java (hunk 9 above)
Comment (caiopavanelli) [reply presto-main/src/main/java/io/prestosql/logging/aspects/UpdateCatalogsFlow.java]: fixed

File: presto-main/src/main/java/io/prestosql/logging/aspects/SqlShowFlowLoggingAspect.java (hunk 10 above)
Comment (caiopavanelli) [reply presto-main/src/main/java/io/prestosql/logging/aspects/SqlShowFlowLoggingAspect.java]: fixed

File: presto-main/src/main/java/io/prestosql/logging/aspects/SqlShowFlowLoggingAspect.java (hunk 11 above)
Comment (caiopavanelli) [reply presto-main/src/main/java/io/prestosql/logging/aspects/SqlShowFlowLoggingAspect.java]: fixed

File: presto-main/src/main/java/io/prestosql/logging/aspects/SqlShowFlowLoggingAspect.java (hunk 12 above)
Comment (caiopavanelli) [reply presto-main/src/main/java/io/prestosql/logging/aspects/SqlShowFlowLoggingAspect.java]: fixed

File: presto-main/src/main/java/io/prestosql/logging/aspects/SqlFlowLoggingAspect.java (hunk 13 above)
Comment (caiopavanelli) [reply presto-main/src/main/java/io/prestosql/logging/aspects/SqlFlowLoggingAspect.java]: fixed

File: presto-main/src/main/java/io/prestosql/logging/aspects/SqlFlowLoggingAspect.java (hunk 14 above)
Comment (caiopavanelli) [reply presto-main/src/main/java/io/prestosql/logging/aspects/SqlFlowLoggingAspect.java]: fixed

File: presto-main/src/main/java/io/prestosql/logging/aspects/SqlFlowLoggingAspect.java (hunk 15 above)
Comment (caiopavanelli) [reply presto-main/src/main/java/io/prestosql/logging/aspects/SqlFlowLoggingAspect.java]: fixed
